x-pack/plugins/search_inference_endpoints/public/components/all_inference_endpoints/render_table_columns/render_actions/actions/copy_id/use_copy_id_action.tsx

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiIcon, EuiTextColor } from '@elastic/eui';
+import React from 'react';
+import * as i18n from '../../../../../../../common/translations';
+import { useKibana } from '../../../../../../hooks/use_kibana';
+import { InferenceEndpointUI } from '../../../../types';
+import { UseCopyIDActionProps } from '../types';
+
+export const useCopyIDAction = ({ onActionSuccess }: UseCopyIDActionProps) => {
```

Comment:
We have built-in EUI hooks for this that you should use instead of creating our own new hook: https://eui.elastic.co/#/utilities/copy

Reply:
Updated. Used EuiCopy instead of custom hook for copy.